From 836270a9a7f1d360a73bc2962287b0bb88add1a4 Mon Sep 17 00:00:00 2001
From: Adrian Alonso <aalonso@freescale.com>
Date: Tue, 27 Oct 2015 16:01:32 -0500
Subject: [PATCH 0998/1691] MLK-11776: pinctrl: freescale: fix default init
 state of conf regs

commit 836270a9a7f1d360a73bc2962287b0bb88add1a4 from
git://git.freescale.com/imx/linux-2.6-imx.git imx_4.1.15_1.0.0_ga

Fix default initialization of iomuxc mux and conf registers,
on parse group pad id relays on dafult init state to set the
corresponding pin_id value for a given pad, if default init
value is missing it can result in erroneous pinctrl configuration.

Signed-off-by: Adrian Alonso <aalonso@freescale.com>
---
 drivers/pinctrl/freescale/pinctrl-imx.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/pinctrl/freescale/pinctrl-imx.c b/drivers/pinctrl/freescale/pinctrl-imx.c
index f3b3827..f3644cf 100644
--- a/drivers/pinctrl/freescale/pinctrl-imx.c
+++ b/drivers/pinctrl/freescale/pinctrl-imx.c
@@ -659,7 +659,7 @@ int imx_pinctrl_probe(struct platform_device *pdev,
 	struct device_node *np;
 	struct imx_pinctrl *ipctl;
 	struct resource *res;
-	int ret;
+	int ret, i;
 
 	if (!info || !info->pins || !info->npins) {
 		dev_err(&pdev->dev, "wrong pinctrl info\n");
@@ -677,6 +677,11 @@ int imx_pinctrl_probe(struct platform_device *pdev,
 	if (!info->pin_regs)
 		return -ENOMEM;
 
+	for (i = 0; i < info->npins; i++) {
+		info->pin_regs[i].mux_reg = -1;
+		info->pin_regs[i].conf_reg = -1;
+	}
+
 	res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
 	ipctl->base = devm_ioremap_resource(&pdev->dev, res);
 	if (IS_ERR(ipctl->base))
-- 
1.9.1

